### DISCUSSION TITLE
U50 JTAG/UART FTDI details

### DISCUSSION THREAD

**Discussion by likewise** [eugene-tarassov/vivado-risc-v#70] [Q&A] [answered]
Hello Eugene,

I noticed you added an example for the U250. I am currently using the U50, and am looking for ways to use the JTAG for monitoring/debugging. Where can we find documentation on the FTDI/JTAG/UART details required to configure for example UART similar to U250?

https://github.com/eugene-tarassov/vivado-risc-v/blob/master/board/u250/uart.xdc

**Comment by eugene-tarassov** [comment, marked as answer]
The [Alveo programming cable](https://www.xilinx.com/products/boards-and-kits/alveo/accessories.html) is required for U50 JTAG and UART access.

UART pins:
FPGA UART TXD - BE26
FPGA UART RXD - BF26
IOSTANDARD LVCMOS18
I'm not sure about direction of the signals.

Documentation on the FTDI/JTAG/UART details not publicly available.

**Comment by likewise** [comment]
I have the cable, wanted to see if a UART could be hooked up. I found three UARTs in the restricted documentation constraints file, but indeed could not find connection details (a block diagram or schematic) beyond the pins. BF26 seems to be input to the FPGA.
